azure/services/asyncpoller/asyncpoller.go

```diff
@@ -73,34 +73,40 @@ func (s *Service[C, D]) CreateOrUpdateResource(ctx context.Context, spec azure.R
 	}
 
-	// Get the resource if it already exists, and use it to construct the desired resource parameters.
-	var existingResource interface{}
-	if existing, err := s.Creator.Get(ctx, spec); err != nil && !azure.ResourceNotFound(err) {
-		errWrapped := errors.Wrapf(err, "failed to get existing resource %s/%s (service: %s)", rgName, resourceName, serviceName)
-		return nil, azure.WithTransientError(errWrapped, getRetryAfterFromError(err))
-	} else if err == nil {
-		existingResource = existing
-		log.V(2).Info("successfully got existing resource", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
-	}
+	// Only when no long running operation is currently in progress do we need to get the parameters.
+	// The polling implemented by the SDK does not use parameters when a resume token exists.
+	var parameters interface{}
+	if resumeToken == "" {
+		// Get the resource if it already exists, and use it to construct the desired resource parameters.
+		var existingResource interface{}
+		if existing, err := s.Creator.Get(ctx, spec); err != nil && !azure.ResourceNotFound(err) {
+			errWrapped := errors.Wrapf(err, "failed to get existing resource %s/%s (service: %s)", rgName, resourceName, serviceName)
+			return nil, azure.WithTransientError(errWrapped, getRetryAfterFromError(err))
+		} else if err == nil {
+			existingResource = existing
+			log.V(2).Info("successfully got existing resource", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
+		}
 
-	// Construct parameters using the resource spec and information from the existing resource, if there is one.
-	parameters, err := spec.Parameters(ctx, existingResource)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get desired parameters for resource %s/%s (service: %s)", rgName, resourceName, serviceName)
-	} else if parameters == nil {
-		// Nothing to do, don't create or update the resource and return the existing resource.
-		log.V(2).Info("resource up to date", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
-		return existingResource, nil
-	}
+		// Construct parameters using the resource spec and information from the existing resource, if there is one.
+		parameters, err = spec.Parameters(ctx, existingResource)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get desired parameters for resource %s/%s (service: %s)", rgName, resourceName, serviceName)
+		} else if parameters == nil {
+			// Nothing to do, don't create or update the resource and return the existing resource.
+			log.V(2).Info("resource up to date", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
+			return existingResource, nil
+		}
 
-	// Create or update the resource with the desired parameters.
-	logMessageVerbPrefix := "creat"
-	if existingResource != nil {
-		logMessageVerbPrefix = "updat"
+		// Create or update the resource with the desired parameters.
+		logMessageVerbPrefix := "creat"
```

Comment:
since we're no longer reusing this, can we simplify the logic and simply print two different log statements if existing resource is nil vs not nil?

Reply:
this would allow us to remove https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.codespellignore#L5

Reply:
Yep, makes sense.

Reply:
Or maybe we can't quite update codespell yet :/
```
/home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/azure/services/aso/aso.go:217: updat ==> update
/home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/azure/services/async/async.go:138: updat ==> update
```